papermill/tests/test_cli.py

```diff
@@ -118,6 +118,15 @@ def test_parameters(self, execute_patch):
             **self.augment_execute_kwargs(parameters={'foo': 'bar', 'baz': 42})
         )
 
+    @patch(cli.__name__ + '.execute_notebook')
+    def test_parameters(self, execute_patch):
+        self.runner.invoke(
+            papermill, self.default_args + ['-p', 'foo', 'bar,baz', '--parameters', 'buzz', '42']
```

Comment:
So I'm a little hesitant on this feature change. An earlier attempt https://github.com/nteract/papermill/pull/411 had to be reverted as it broke other CLI interactions. Specifically my concern for this PR is that many users pass JSON as parameter inputs: e.g. `-p '["foo","bar"]'`. Another issue would be that this would currently block users from passing "bar,baz" if they needed to as a string. It might be such that we want to not approach this problem in the CLI layer itself and leave yaml/json inputs to handle list/map types. Open to conversation though if you see a clean way past concerns.

Reply:
Totally understand the concerns. My thinking was definitely that the YAML/JSON inputs would be better suited for inputting lists. I was just going off the ticket. I don't really think there's a clean way to do this without breaking those use cases. The only thought would be adding a flag to enable/disable this functionality but I wouldn't exactly say that's clean. That seems like a confusing workaround.

Reply:
Yeah, I missed closing the original issue when reverted the other attempt. It can get pretty gnarley to handle all the cases correctly while maintaining a simple interface with `-p`options (if it's even possible to do so).

Reply:
@MSeal Maybe some additional examples in the documentation on how to pass in different types of parameters would help users? I could add a section specifically on "complex types" focusing on the options for passing in lists/structs.

Reply:
That'd be awesome, if you wanted to make a docs PR!